misc/metrics/src/swarm.rs

```diff
@@ -261,25 +296,42 @@ impl<TBvEv, THandleErr> super::Recorder<libp2p_swarm::SwarmEvent<TBvEv, THandleE
             libp2p_swarm::SwarmEvent::ListenerError { .. } => {
                 self.listener_error.inc();
             }
-            libp2p_swarm::SwarmEvent::Dialing(_) => {
+            libp2p_swarm::SwarmEvent::Dialing { .. } => {
                 self.dial_attempt.inc();
             }
         }
     }
 }
 
 #[derive(EncodeLabelSet, Hash, Clone, Eq, PartialEq, Debug)]
-struct ConnectionEstablishedLabels {
+struct ConnectionLabels {
     role: Role,
     protocols: String,
 }
 
-type ConnectionEstablishmentDurationLabels = ConnectionEstablishedLabels;
-
 #[derive(EncodeLabelSet, Hash, Clone, Eq, PartialEq, Debug)]
 struct ConnectionClosedLabels {
-    role: Role,
-    protocols: String,
+    // TODO: Should be Option<ConnectionError>. Needs https://github.com/prometheus/client_rust/pull/137
```

Comment:
Do we wait for this before merging here?

Reply:
Oversight. Fixed with [9578315](https://github.com/libp2p/rust-libp2p/pull/3927/commits/9578315e6409a50693ab1ec746fd2c6b589f5166). Thanks for the catch Thomas!